src/main/java/edu/byu/cs/autograder/QualityAnalyzer.java

```diff
@@ -66,67 +74,100 @@ private void parseOutput(String output) {
                 warnings.add(trimmed);
             }
         }
+        return new QualityOutput(errors, warnings);
     }
 
-    private float getScore() {
+    /**
+     * Evaluates the score for a quality analysis
+     *
+     * @param output errors and warnings from quality analysis
+     * @return the score as a float from 0 to 1
```

Comment:
Ooh good point. Knowing the range of the value is really helpful. I nearly forgot that would be a tricky point moving forward!